File: NOVEMBER-2025/24-11-2025/1018.rb
# @param {Integer[]} nums
# @return {Boolean[]}
def prefixes_div_by5(nums)
    ans = []
    mod = 0

    nums.each do |bit|
        mod = (mod * 2 + bit) % 5
        ans << (mod == 0)
    end

    ans
end

File: NOVEMBER-2025/24-11-2025/1018.ts
function prefixesDivBy5(nums: number[]): boolean[] {
    const ans: boolean[] = [];
    let mod = 0;

    for (const bit of nums) {
        mod = (mod * 2 + bit) % 5;
        ans.push(mod === 0);
    }

    return ans;
}